biglog/segment.go

```diff
@@ -164,6 +173,9 @@ func loadSegment(indexPath string) (seg *segment, err error) {
 	seg.setNextOffsets()
 	seg.setCreatedTS()
 
+	// TODO: when this is called via createSegment the index is
+	// created by createSegIndex which doesn't write this timestamp?!
```

Comment:
a timestamp in the index signals that there is data, updating it without data might break a few places.

currently the indexes are pre-allocated, so there is no danger of the index having different capacity, but having a header on the segment might be indeed useful and not too disrupting since in the index you can define the data file offset of the first entry to the length of the header and everything should work the same.